ipamutils/utils.go

```diff
@@ -115,7 +115,7 @@ func splitNetwork(size int, base *net.IPNet) []*net.IPNet {
 
 	for i := 0; i < n; i++ {
 		ip := copyIP(base.IP)
-		addIntToIP(ip, uint(i<<s))
+		addIntToIP(ip, uint(i), s)
```

Comment:
If i understand the problem here, for ipv6 i << s may be larger than (1<<64)-1; i wonder if we should instead shift ordeal into an byte array to accommodate that. It definitely making the change a lot more understandable?